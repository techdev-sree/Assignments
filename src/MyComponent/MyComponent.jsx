import React, { Component } from 'react';
import { sayHello } from '../utils/utils';
import styles from './styles.modules.scss';

class MyComponent extends Component {
    constructor(props) {
        super(props);
        this.state = {
            name: 'SFO',
            rendered: false
        };
    }

    componentDidMount() {
        this.setState({ rendered: true });
    }

    render() {
        return (
            <div className={styles.myComponent}>
                {sayHello(this.state.name)}
            </div>
        );
    }
}

export { MyComponent };