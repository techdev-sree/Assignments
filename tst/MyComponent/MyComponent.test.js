import React from 'react';
import { MyComponent } from '../../src/MyComponent/MyComponent';
import { shallow } from 'enzyme';

describe('MyComponent Test', () => {
    it('render correctly text component', () => {
        const myComponent = shallow(<MyComponent/>);
        expect(myComponent).toMatchSnapshot();
        expect(myComponent.state()).toEqual({ name: 'SFO', rendered: true });
        expect(myComponent.state('rendered')).toBeTruthy();
    });
});
