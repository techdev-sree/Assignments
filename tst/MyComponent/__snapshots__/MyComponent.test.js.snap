// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent Test render correctly text component 1`] = `ShallowWrapper {}`;
